### ISSUE TITLE
git_submodule_path does not return correct path

### ISSUE THREAD

**Issue by csware** [libgit2/libgit2#3365]
Take the repository https://github.com/mpc-hc/mpc-hc and use `git_submodule_foreach` to iterate over the submodules.  And now notice that inside the callback function the function `git_submodule_path` for the submodule with the following definition returns `LAVFilters` instead of `src/thirdparty/LAVFilters/src`.

```
[submodule "LAVFilters"]
    path = src/thirdparty/LAVFilters/src
    url = https://github.com/mpc-hc/LAVFilters.git
```

Fun fact:
It works correctly for the following submodules:
```
[submodule "bs2b"]
    path = src/thirdparty/bs2b/libbs2b
    url = git://github.com/alexmarsev/libbs2b.git
[submodule "lcms2"]
    path = src/thirdparty/lcms2/library
    url = https://github.com/mm2/Little-CMS.git
[submodule "MediaInfoLib"]
    path = src/thirdparty/MediaInfo/library
    url = https://github.com/MediaArea/MediaInfoLib.git
[submodule "soxr"]
    path = src/thirdparty/soxr/libsoxr
    url = git://github.com/alexmarsev/soxr.git
[submodule "ZenLib"]
    path = src/thirdparty/ZenLib/library
    url = https://github.com/MediaArea/ZenLib.git
```

Other failing submodules are:
```
[submodule "RARFileSource"]
    path = src/thirdparty/RARFileSource/library
    url = https://github.com/mpc-hc/rarfilesource.git
[submodule "soundtouch"]
    path = src/thirdparty/SoundTouch/soundtouch
    url = git://github.com/alexmarsev/soundtouch.git
[submodule "sanear"]
    path = src/thirdparty/sanear/sanear
    url = git://github.com/alexmarsev/sanear.git
    branch = core
```

**Comment by csware** [comment]
It seems as if the `map`in `submodule.c` isn't updated with the new key.

**Comment by carlosmn** [comment]
Fixed via #3370
